engine/pom.xml

```diff
@@ -218,8 +218,10 @@
                             org.hibernate.search.query;version="${project.version}",
                             org.hibernate.search.query.collector;version="${project.version}",
                             org.hibernate.search.query.dsl;version="${project.version}",
+                            org.hibernate.search.query.dsl.sort;version="${project.version}",
                             org.hibernate.search.query.engine;version="${project.version}",
                             org.hibernate.search.query.engine.spi;version="${project.version}",
+                            org.hibernate.search.query.engine.spi.impl;version="${project.version}",
```

Comment:
Well, in this case, it's an implementation of an SPI, so leakage is necessary. Actually that's the whole point of an SPI: providing a way for others to expose another implementation.

In this particular case, it's exported because it's used there: https://github.com/yrodiere/hibernate-search/blob/HSEARCH-1872/orm/src/main/java/org/hibernate/search/impl/FullTextSessionImpl.java#L87

Originally, the class in this `spi.impl` package was part of `ORM`. But:

 1. It has nothing to do with ORM, but rather with the Lucene backend
 2. More importantly, QueryDescriptors are the easiest way to create queries in a backend-independant way. I needed this for implementing a backend-independant test in -engine. See https://github.com/yrodiere/hibernate-search/blob/HSEARCH-1872/engine/src/test/java/org/hibernate/search/test/dsl/SortDSLTest.java#L643 in particular.

Now, while the leakage of an SPI implementation is no big deal in my opinion, a better solution certainly exists. I'd be in favor of moving the code in https://github.com/yrodiere/hibernate-search/blob/HSEARCH-1872/orm/src/main/java/org/hibernate/search/impl/FullTextSessionImpl.java#L87 up to the -engine module, since it actually implements an -engine feature: the ability to use a different backend for creating a `HSQuery`. I'm a bit lost as where to expose this feature, though... I can try to find some place if you think it's the way to go.